lib/wibox/layout/grid.lua

```diff
@@ -0,0 +1,712 @@
+                    --"max_cols_size", "max_rows_size",
+                   }
+
+local dir_properties = { "spacing", "padding", "fixed_size", "expand" }
+
+
+--- Reset a ratio layout. This removes all widgets from the layout.
+-- **Signal:** widget::reset
+-- @param layout The layout you are modifying.
+-- @name reset
+-- @class function
+
+
+--- The preferred orientation for the grid.
+-- @beautiful beautiful.grid_orientation
+
+--- The spacing between cells.
+-- @beautiful beautiful.grid_[horizontal|vertical]_spacing
+
+--- The padding with outer limits.
+-- @beautiful beautiful.grid_[horizontal|vertical]_padding
+
+--- Controls if the columns/rows are expanded to fit all the available area.
+-- @beautiful beautiful.grid_[horizontal|vertical]_expand
+
+--- Controls if the columns/rows have the same size.
+-- @beautiful beautiful.grid_[horizontal|vertical]_fixed_size
+
+
+--- The spacing between cells.
+-- If direction is not specified, set both directions and get the preferred orientation
+--@DOC_wibox_layout_grid_spacing_EXAMPLE@
+-- @property [horizontal_|vertical_|]spacing
+
+--- The padding with outer limits.
+-- If direction is not specified, set both directions and get the preferred orientation
+--@DOC_wibox_layout_grid_spacing_EXAMPLE@
+-- @property [horizontal_|vertical_|]padding
+
+--- Controls if the columns/rows are expanded to fit all the available area.
+-- If direction is not specified, set both directions and get the preferred orientation
+--@DOC_wibox_layout_grid_expand_EXAMPLE@
+-- @property [horizontal_|vertical_|]expand
+
+--- Controls if the columns/rows have the same size.
+-- If direction is not specified, set both directions and get the preferred orientation
+--@DOC_wibox_layout_grid_expand_EXAMPLE@
+-- @property [horizontal_|vertical_|]fixed_size
+
+--- The minimum size for the columns
+-- @property min_cols_size
+
+--- The minimum size for the rows
+-- @property min_rows_size
+
+
+
+-- Return the maximum value of a table
+local function max_value(t)
+    local m = 0
+    for _,v in ipairs(t) do
+        if m < v then m = v end
+    end
+    return m
+end
+
+
+--- Add a widget to the matrix
+-- the widget is set in coordinates `row`:`col`, spanning several cells
+-- given by `row_span` and `col_span`
+-- @param matrix Matrix containing the widgets
+-- @param widget Widget to add to the matrix
+-- @param number row Row number for the top left corner of the widget
+-- @param number col Column number for the top left corner of the widget
+-- @param number row_span The number of rows the widget spans (default to 1)
+-- @param number col_span The number of columns the widget spans (default to 1)
+-- @treturn boolean index If the operation is successful
+local function matrix_add(matrix, object, row, col, row_span, col_span)
+    if row < 1 then return false end
+    if col < 1 then return false end
+    row_span = row_span > 0 and row_span or 1
+    col_span = col_span > 0 and col_span or 1
```

Comment:
Well, it's a security in case the code is updated and this part is used in some other place. It could be removed, but is it really a problem?